src/Instrumentation/Event.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class EventWithDuration.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Instrumentation;
+
+/**
+ * A server-timing event.
+ *
+ * @package AmpProject\AmpWP
+ */
+class Event {
+
+	/**
+	 * Event name.
+	 *
+	 * @var string
+	 */
+	protected $name;
+
+	/**
+	 * Event description.
+	 *
+	 * @var string|null
+	 */
+	protected $description;
+
+	/**
+	 * Additional properties of the event.
+	 *
+	 * @var string[]
+	 */
+	protected $properties;
+
+	/**
+	 * Event constructor.
+	 *
+	 * @param string        $name        Event name.
+	 * @param string|null   $description Optional. Event description.
+	 * @param string[]|null $properties  Optional. Additional properties for the
+	 *                                   event.
+	 */
+	public function __construct( $name, $description = null, $properties = [] ) {
+		$this->name        = $name;
+		$this->description = $description;
+		$this->properties  = (array) $properties;
+	}
+
+	/**
+	 * Get the name of the event.
+	 *
+	 * @return string Event name.
+	 */
+	public function get_name() {
+		return $this->name;
+	}
+
+	/**
+	 * Get the description of the event.
+	 *
+	 * @return string Event description.
+	 */
+	public function get_description() {
+		return $this->description ?: '';
+	}
+
+	/**
+	 * Add additional properties to the event.
+	 *
+	 * @param string[] $properties Properties to add.
+	 */
+	public function add_properties( $properties ) {
+		$this->properties = array_merge(
+			(array) $properties,
```

Comment:
Yes, as this might be triggered from different contexts, I wanted to ensure that one part of the code can't override and undo another part of the code, this is "add-only".

Yes, I can add that verification.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/5005/commits/c84ab1166cacee62febe2eaac629192daf5689d4